### PR TITLE
Add error handling, clean up search term parsing, and add comments

### PR DESCRIPTION
**Pull request by mattgawarecki**
I'm no CoffeeScript programmer, but I noticed it could be confusing when Giphy doesn't return results for a given search term. I was also perhaps misguided in the idea that quoting search terms would lead to more accurate results, but so far it hasn't hurt anything. Feel free to change this as you see fit, but I think the error handling could be useful, and the comments may be useful to any future contributors.

I also created a separate respond hook for a simple "random" gif me request without any terms -- that way, the logic to determine if you've got search terms or not becomes unnecessary.

Hopefully you find use in these changes! Whether or not you merge them, it was fun to work on. My co-workers very much enjoy this Hubot plugin, so keep up the good work.

### CODE REVIEW

File: src/giphy-gifme.coffee [hunk 1]
```diff
@@ -9,25 +9,95 @@
...
+#       This effectively passes the problem on to the Giphy API, while still offering some basic
+#       parsing work before it gets to that stage.
+getTags = (raw_tag_string) ->
+  # Create a new string for us to sanitize, transform, and munge out search tags
+  workingTagString = trim(raw_tag_string)
+
+  # Start with an empty array of tags
+  tags = []
+
+  # Seach for quoted strings
```
Comment (mattgawarecki) [inline src/giphy-gifme.coffee]: Feel free to correct this typo... d'oh.

File: src/giphy-gifme.coffee [hunk 2]
```diff
@@ -9,25 +9,95 @@
...
+# hubot gif me tag1 tag2 - Search for a GIF tagged with "tag1" and "tag2"
 #
 # Author:
 # Ben Centra
 
-api_key = process.env.HUBOT_GIPHY_API_KEY or 'dc6zaTOxFJmzC' # <== Giphy's public API key, please request your own!
+# Giphy's public API key; please request your own!
+api_key = process.env.HUBOT_GIPHY_API_KEY or 'dc6zaTOxFJmzC'
+
+# Content rating limit:
```
Comment (mattgawarecki) [inline src/giphy-gifme.coffee]: Forgot to add valid content ratings to the comment on line 23... just a little too hasty with this PR, I think. If you'd like to correct it, valid values are: y,g, pg, pg-13 or r

**Comment by bencentra** [comment]
Sorry for overlooking this for so long! I'm gonna get this up-to-date with the latest changes and merge it ASAP.

**Comment by bencentra** [comment]
Closing this PR. Much of it was used as feedback in the last few "refactor" commits and released as v0.1.7: https://github.com/bencentra/hubot-giphy-gifme/releases/tag/v0.1.7.

**Pull request closed without merging**